services/decoder/protocols/lido/lido.decoders.ts

```diff
@@ -0,0 +1,881 @@
+        };
+
+        const details: EventDetails = [
+            {
+                heading: "Sender",
+                value: decoded.sender,
+                type: "address",
+            },
+            {
+                heading: "To",
+                value: decoded.amount.toString(),
+                type: "text",
+            },
+            {
+                heading: "Referral",
+                value: decoded.referral,
+                type: "address",
+            },
+        ];
+
+        const tokens: EventTokens = [
+            {
+                heading: "Tokens Staked",
+                value: tx?.value!.toString(),
+                decimals: tx?.gas_metadata?.contract_decimals,
+                ticker_symbol: tx?.gas_metadata?.contract_ticker_symbol,
+                ticker_logo: tx?.gas_metadata?.logo_url,
+                pretty_quote: tx?.value_quote?.toString(),
+            },
+        ];
+
+        return {
+            action: DECODED_ACTION.TRANSFERRED,
+            category: DECODED_EVENT_CATEGORY.STAKING,
+            name: "Submitted",
+            protocol: {
+                logo: log_event.sender_logo_url as string,
+                name: "Lido" as string,
+            },
+            details,
+            tokens,
+        };
+    }
+);
+
+GoldRushDecoder.on(
+    "lido:TokenRebased",
+    ["eth-mainnet"],
+    ABI as Abi,
+    async (log_event, tx, chain_name, covalent_client): Promise<EventType> => {
+        const { raw_log_data, raw_log_topics } = log_event;
+
+        const { args: decoded } = decodeEventLog({
+            abi: ABI,
+            topics: raw_log_topics as [],
+            data: raw_log_data as `0x${string}`,
+            eventName: "TokenRebased",
+        }) as {
+            eventName: "TokenRebased";
+            args: {
+                reportTimestamp: bigint;
+                timeElapsed: bigint;
+                preTotalShares: bigint;
+                preTotalEther: bigint;
+                postTotalShares: bigint;
+                postTotalEther: bigint;
+                sharesMintedAsFees: bigint;
+            };
+        };
+
+        const details: EventDetails = [
+            {
+                heading: "Time Stamp",
+                value: decoded.reportTimestamp.toString(),
+                type: "text",
+            },
+            {
+                heading: "Time Elapsed",
+                value: decoded.timeElapsed.toString(),
+                type: "text",
+            },
+            {
+                heading: "Pre-Total Shares",
+                value: decoded.preTotalShares.toString(),
+                type: "text",
+            },
+            {
+                heading: "Post-Total Shares",
+                value: decoded.postTotalShares.toString(),
+                type: "text",
+            },
+            {
+                heading: "Shares Minted As Fees",
+                value: decoded.postTotalShares.toString(),
+                type: "text",
+            },
```

Comment:
Parse all the time options to human readable strings, maybe use `timestampParser`

Reply:
Left time elapsed because that is supposed to represent the number of seconds since the last event but converted all the others.